src/Simulation/Core/ArgumentType.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// The type of an argument to a QIR callable.
+    /// </summary>
+    public class ArgumentType
```

Comment:
What is the idea behind having this be part of the runtime package rather than the entry point driver (holds for Argument.cs, ArgumentType.cs and ArgumentValue.cs)? I assume they are public (only) so the entry point driver can access them? 
It's confusing to me that this is a data type in the Q# runtime called ArgumentType that does not contain an entry for all possible argument types in Q#.

Reply:
The reason for having this class in the runtime package is that it can be used by the specific client that submits the program to Azure as well as the entry point driver.

Yes, the `ArgumentType` class supports the subset of Q# types currently supported as entry-point arguments to be submitted to Azure Quantum, so even though it is confusing it is "by design".

Reply:
It would be a good idea to have a conversation about this before merging since in its current implementation this class needs to be `public`.

Reply:
Discussed offline. While we know this is going to change in the near future, we don't really have an option but to have the public class matching the bond schema. I am fine with that.